CMakeLists.txt

```diff
@@ -94,7 +94,11 @@ else()
   if(WITH_ZLIB)
     find_package(zlib REQUIRED)
     add_definitions(-DZLIB)
-    include_directories(${ZLIB_INCLUDE_DIR})
+    if(CMAKE_VERSION VERSION_LESS "3.0")
```

Comment:
@xkszltl i'd suggest avoid checking the `CMAKE_VERSION` unless we have no other options. as it's a magic number in this context. how about just checking if `ZLIB_INCLUDE_DIRS` is available, and use `ZLIB_INCLUDE_DIR` as a fallback? like
```cmake
if(ZLIB_INCLUDE_DIRS)
  include_directories(${ZLIB_INCLUDE_DIRS})
else()
  include_directories(${ZLIB_INCLUDE_DIR})
endif()
```

and Happy New Year!